webknossos/webknossos/geometry/vec_int.py

```diff
@@ -0,0 +1,283 @@
+import re
+from operator import add, floordiv, mod, mul, sub
+from typing import Any, Callable, Iterable, List, Optional, Tuple, Union, cast
+
+import numpy as np
+
+VALUE_ERROR = "VecInt can be instantiated with int values `VecInt(1,2,3,4) or with `VecIntLike` object `VecInt([1,2,3,4])."
+
+
+class VecInt(tuple):
+    """
+    Class to represent a ND vector of integer values.
+    """
+
+    def __new__(
+        cls,
+        vec: Union[int, "VecIntLike"],
+        *args: int,
+        y: Optional[int] = None,
+        z: Optional[int] = None,
+        **kwargs: int,
+    ) -> "VecInt":
+        """
+        Class to represent a ND vector. Inherits from tuple and provides useful
+        methods and operations on top. The vector has a minimal length of 3.
+
+        A small usage example:
+
+        ```python
+        from webknossos import VecInt
+
+        vector_1 = VecInt(1, y=2, z=3, t=4)
+        vector_2 = VecInt.full(1, 4)
+        assert vector_2[0] == vector_2[1] == vector_2[2] == vector_2[3]
+
+        assert vector_1 + vector_2 == (2, 3, 4, 5)
+        ```
+        """
+
+        if isinstance(vec, VecInt):
+            return vec
+
+        as_tuple: Optional[Tuple[int, ...]] = None
+
+        if isinstance(vec, int):
+            if args:
+                assert all(isinstance(arg, int) for arg in args), VALUE_ERROR
+                as_tuple = (vec, *args)
+            else:
+                assert y is not None and z is not None, VALUE_ERROR
+                remaining_axes = []
+                for key, value in kwargs.items():
+                    assert key not in [
+                        "x",
+                        "y",
+                        "z",
+                    ], "Additional axis must not be called x, y or z."
```

Comment:
I just thought about it again. And since the VecInt class currently does not save axis names, it is quite misleading to implement the naming of the axes with kwargs at all. 

I currently see two possibilities: Either I implement axes names on VecInt or I change the signature of the VecInt class to:
```python
def __new__(
    cls,
    vecOrInt: Union[int, "VecIntLike"],
    *args: int,
) -> "VecInt":
```
An argument for implementing the axes names would be to enable access to `x`, `y` and `z` property again, at least if they are present. 

On the other hand, instantiation would be more complex, because there is no default order for all axes, as in the Vec3Int class. I could not do `VecInt(1,2,3,4)` anymore. Operation on axes aware VecInts would be more complex too, because I can't apply operators for elements with the same index. Instead, a compatibility check is necessary (axes order is the same) or look up which axes of one VecInt corresponds to which of the other.

Reply:
I implemented the VecInt class with axes names now and removed the check you mentioned here. I decided to not use the signature you provided, as vecOrX would be a keyword argument, while `*args` are positional arguments. A keyword argument is not allowed in front of them. I removed the vecOrX completely and parsed all arguments from the signature:
```python
def __new__(
        cls,
        *args: Union["VecIntLike", Iterable[str], int],
        axes: Optional[Iterable[str]] = None,
        **kwargs: int,
    ) -> "VecInt"
```

To create a vector with 4 elements and axes x, y, z, t:
```python
        vector_1 = VecInt(1, 2, 3, 4, axes=("x", "y", "z", "t"))
        # or
        vector_1 = VecInt([1, 2, 3, 4], axes=("x", "y", "z", "t"))
        # or
        vector_1 = VecInt(x=1, y=2, z=3, t=4)
```

Reply:
sounds great :+1: